skimage/filters/_gaussian.py

```diff
@@ -121,8 +121,8 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
         if not isinstance(output, np.ndarray):
             raise ValueError("Provided output argument is not a numpy array")
         elif image.dtype != output.dtype:
-            msg = ("Invalid output datatype. A new array is assigned.")
-            output = np.empty_like(image)
+            msg = ("Invalid output datatype. A new array wiil be assigned.")
+            output = None
```

Comment:
This will not solve the CI failures with scipy 0.19: In this case if output is None you can directly return ndi.gaussian_filter(image, **kwargs) as it is done now but when output is set to an array, you have to return output.

Reply:
An other option would be to set output to an empty array if it is None, call ndi.gaussian_filter and then return output.

Reply:
This doesn't solves CI failures with scipy 0.19, it just transfers the initialization of `output` to the `ndi.gaussian_filter`.

Reply:
This is true, but the CI still fails :-)

Reply:
It seems like for different versions of `scipy` `ndi.gaussian_filter` has different behavior, when `output` is `None`.

Reply:
I have made update, if will not work, I will return `output`.

Reply:
Unfortunately, CI with fail when scipy version is 0.19... You are right, `ndi.gaussian_filter` behavior changed with the more recent versions of scipy.

Reply:
Have changed, now it returns output. But I think this may cause the other tests to crash.

Reply:
The CI failure is not caused by your changes. IMHO every thing is fine ;-) Good job!

Reply:
Thank you! I was confused with `gaussian skimage.filters.gaussian` input requirements and `ndi.ndimage.gaussian_filter` different versions behavoiur. So it took more time than I expected to fix this bug.
Anyway bug is fixed, docstrings and tests are added :)

Reply:
Thank you :)